### PR TITLE
Add status colours for individual services

### PR DESCRIPTION
**Pull request by FiniteReality**
After reading https://github.com/dotnet/tye/issues/443#issuecomment-624835769, I was slightly inspired to add something to indicate whether the overall set of services are both healthy and ready.

Currently, the colours are as follows, in order of priority:
- All `ingress` services are colourless, as their replicas do not appear to have states.
- A service where any replica is in the `Removed`, `Added`, `Started` or `Stopped` state is coloured red.
- A service where any replica is in the `Healthy` state is coloured yellow.
- A service whose replicas all are in the `Ready` state are coloured green.

I haven't been able to check any azure web functions yet since those cause errors... (Apparently I'm missing the tools, even though I installed them?)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=681) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=681) before we can accept your contribution.<br/><br/>:x: FiniteReality [sign now](https://cla.dotnetfoundation.org/dotnet/tye?pullRequest=681)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/tye?pullRequest=681) it.</sub>

File: src/Microsoft.Tye.Hosting/Dashboard/Pages/Index.razor [hunk 1]
```diff
@@ -96,6 +97,37 @@
         return $"{(b.Protocol ?? "tcp")}://{b.Host ?? "localhost"}:{b.Port}";
     }
 
+    string GetServiceStatusColor(Service service)
```
Comment (davidfowl) [inline src/Microsoft.Tye.Hosting/Dashboard/Pages/Index.razor:100]: nit: Use braces even for single line statements

**Comment by davidfowl** [comment]
Screenshot please

**Comment by FiniteReality** [comment]
Here's some screenshots using the liveness-and-readiness sample:

Healthy and ready:
Screenshot_20200928_220903
Healthy:
Screenshot_20200928_220951
Neither healthy nor ready: 
Screenshot_20200928_221014

**Comment by davidfowl** [comment]
I think it should be a checkmarks instead of changing the entire background color

**Comment by jkotalik** [comment]
Or just directly reporting healthy/ready in another column?

**Comment by FiniteReality** [comment]
> Or just directly reporting healthy/ready in another column?

I went with colour mainly because I felt it was easier to see the overall status at a glance. Checkmarks in their own column may work, however.

**Comment by FiniteReality** [comment]
Here are some updated screenshots:

Healthy and ready:
Screenshot_20201003_212623
Healthy: 
Screenshot_20201003_212704
Neither healthy nor ready:
Screenshot_20201003_212741

Ideally I wouldn't have included the entire set of bootstrap icons, but I was having issues getting colours to work so I decided it would likely be for the best.

The bootstrap icons are under the MIT license, but I currently haven't included any copyright notice since I didn't know if this would be ideal, or where the most appropriate place to put the notice would be. For reference, here's the license link: https://github.com/twbs/icons/blob/main/LICENSE.md

**Comment by jkotalik** [comment]
@Pilchie to use these icons, do we need to include the license in our own LICENSE.txt file?

Besides that, this LGTM.

**Comment by Pilchie** [comment]
So, the license is the same as the license in our repo right? I think what we need would be an entry in a 3rd party notice to give credit.  Will send some internal email on processes for this.

**Comment by JunTaoLuo** [comment]
@Pilchie I couldn't find an email regarding this. Do we just need to add a 3PN like https://github.com/dotnet/aspnetcore/blob/master/THIRD-PARTY-NOTICES.txt in this repo or is there something else we need to do?

**Comment by jkotalik** [comment]
Should just need the 3PN. I'll update this PR now.

**Comment by jkotalik** [comment]
Closing and reopening with a third party notice. Thanks @FiniteReality

**Pull request closed without merging**